src/lib.rs

```diff
@@ -538,3 +540,53 @@ impl Window {
         );
     }
 }
+
+/// A convenient result type consisting of a return type and a [`Error`].
+pub type QuadResult<T = ()> = Result<T, Error>;
+
+/// Global `macroquad` error.
+///
+/// This error contains all `macroquad` error types. You can use this for all `macroquad` functions that can return `Result`.
+/// ```rust ignore
+/// #[macroquad::main("Error")]
+/// async fn main() -> QuadResult {
+///     let texture = load_texture(/* ... */).await?;
+///     let material = load_material(/* ... */)?;
+///     // ...
+/// }
+/// ```
+#[derive(Debug)]
+pub enum Error {
```

Comment:
While we use `QuadResult` notation - I think it is reasonable to call an error `QuadError`